### ISSUE TITLE
Can't set default options in ConfigurableModuleBuilder setExtras

### ISSUE THREAD

**Issue by vlototskyi** [nestjs/nest#10307]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

```
import { BigQueryOptions } from '@google-cloud/bigquery';
import { ConfigurableModuleBuilder } from '@nestjs/common';

export const { ConfigurableModuleClass, MODULE_OPTIONS_TOKEN } =
  new ConfigurableModuleBuilder<BigQueryOptions>()
    .setExtras<{ isGlobal?: boolean }>(
      { isGlobal: true },
      (definition, extras) => {
        console.log('extras', extras);
        return { ...definition, global: extras.isGlobal };
      },
    )
    .setClassMethodName('forRoot')
    .build();

```

According to documentation: "Values you specify within the "extras" object will be used as default values (first argument of setExtras method)". In practice, default values ​​is missing in the function callback.

For example, in the code above, the default value for isGlobal is true, but in practice it is not in the callback and the module does not behave as global.

### Minimum reproduction code

https://github.com/vlototskyi/nestjs-configurable-module-builder-default-options

### Steps to reproduce

1. `npm install`
2. `npm run start`
3. See logs and error

### Expected behavior

setExtra method will return default options in callback extra parameter, as specified in documentation.

### Package

- [ ] I don't know. Or some 3rd-party package
- [X] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

9.1.1

### Packages versions

[Nest CLI]
Nest CLI Version : 9.1.3 

[Nest Platform Information]
platform-express version : 9.1.1
schematics version           : 9.0.3
common version                : 9.1.1
core version                       : 9.1.1
cli version                           : 9.1.3

### Node.js version

16.15.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

### Other

<img width="1597" alt="Screenshot 2022-09-21 at 18 07 24" src="https://user-images.githubusercontent.com/52038511/191541184-f67cbaaf-6f0b-48df-b4f9-37b68173c9e8.png">

**Comment by micalevisk** [comment]
![image](https://user-images.githubusercontent.com/13461315/191548186-d074ac94-2702-49bf-a219-87afab52b260.png)

you need to import the module that exports that provider.

**Comment by kamilmysliwiec** [comment]
Fixed in 9.1.2
